pkgs/applications/virtualization/anbox/default.nix

```diff
@@ -0,0 +1,176 @@
+    owner = "anbox";
+    repo = "anbox-modules";
+    rev = "5635fc41a23ba415af41f80d12251a74f91ffbb6";
+    sha256 = "0gds35svblkrgynii8j5b9mfm5n8r9rq8ddmx38v5z7qlmdnlr6y";
+  };
+  meta = with stdenv.lib; {
+    homepage = "http://anbox.io";
+    description = "Container based approach to boot a full Android system";
+    license = licenses.gpl3;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.bkchr ];
+  };
+in {
+  ashmem = stdenv.mkDerivation {
+    name = "anbox-ashmem-${version}";
+    inherit meta;
+    src = modules_src;
+    sourceRoot = "source/ashmem";
+    hardeningDisable = [ "pic" ];
+    makeFlags = "KERNEL_SRC=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build DESTDIR=$(out)";
+    nativeBuildInputs = kernel.moduleBuildDependencies;
+    postInstall = ''
+      mkdir -p $out/lib/modules/${kernel.modDirVersion}/extra/anbox
+      mv $out/ashmem_linux.ko $out/lib/modules/${kernel.modDirVersion}/extra/anbox/ashmem.ko
+    '';
+  };
+
+  binder = stdenv.mkDerivation {
+    name = "anbox-binder-${version}";
+    inherit meta;
+    src = modules_src;
+    sourceRoot = "source/binder";
+    hardeningDisable = [ "pic" ];
+    makeFlags = "KERNEL_SRC=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build DESTDIR=$(out)";
+    nativeBuildInputs = kernel.moduleBuildDependencies;
+    postInstall = ''
+      mkdir -p $out/lib/modules/${kernel.modDirVersion}/extra/anbox
+      mv $out/binder_linux.ko $out/lib/modules/${kernel.modDirVersion}/extra/anbox/binder.ko
+    '';
+  };
+
+  anbox = stdenv.mkDerivation {
+    name = "anbox-${version}";
+    inherit src meta;
+
+    nativeBuildInputs = [ cmake pkgconfig ];
+
+    buildInputs = [
+      boost SDL2 mesa_glu glm protobuf dbus-cpp lxc
+      gtest xorg.libpthreadstubs xorg.libXdmcp SDL2_image
+      makeWrapper
+    ];
+
+    preConfigure = ''
+      # Don't build tests
+      truncate -s 0 cmake/FindGMock.cmake
+      truncate -s 0 tests/CMakeLists.txt
+
+      substituteInPlace scripts/gen-emugl-entries.py --replace '/usr/bin/env python2' "${python2}/bin/python"
+    '';
+    cmakeFlags = "-DCMAKE_INSTALL_LIBDIR=lib";
+
+    postInstall = ''
+      mkdir -p $out/share/dbus-1/services/
+      cat <<END > $out/share/dbus-1/services/org.anbox.service
+      [D-BUS Service]
+      Name=org.anbox
+      Exec=$out/libexec/anbox-session-manager
+      END
+
+      mkdir $out/libexec
+      cat > $out/libexec/anbox-session-manager <<EOF
+      #!${bash}/bin/bash
+      exec $out/bin/anbox session-manager
+      EOF
+      chmod +x $out/libexec/anbox-session-manager
+
+      cat > $out/bin/anbox-application-manager <<EOF
+      #!${bash}/bin/bash
+      ${systemd}/bin/busctl --user call \
+         org.freedesktop.DBus \
+         /org/freedesktop/DBus \
+         org.freedesktop.DBus \
+         StartServiceByName "su" org.anbox 0
+
+      $out/bin/anbox launch --package=org.anbox.appmgr --component=org.anbox.appmgr.AppViewActivity
+      EOF
```

Comment:
You can use makeWrapper both scripts here. It also has a `--run` parameter for the busctl command.